File: Redis_Gradle_Netty/src/main/java/course/thrift/simplecase/PersonServiceImpl.java
package course.thrift.simplecase;

import course.thrift.java.generated.DataException;
import course.thrift.java.generated.Person;
import course.thrift.java.generated.PersonService;
import org.apache.thrift.TException;

public class PersonServiceImpl implements PersonService.Iface{
    @Override
    public Person getPersonByUsername(String username) throws DataException, TException {
        System.out.println("Got Client Params: " + username);

        Person person = new Person();
        person.setUsername(username);
        person.setAge(20);
        person.setMarried(false);

        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {
        System.out.println("Got Client Params: " + person);
    }
}

File: Redis_Gradle_Netty/PythonHome/thrift/PersonServiceImpl.py
# _*_ coding:utf-8 _*_
__author__ = 'Jingxuan Ai'

from course.thrift.py.generated import ttypes

class PersonServiceImpl:

    def getPersonByUsername(self, username):
        print("Got client params: " + username)

        person = ttypes.Person()
        person.username = username
        person.age = 30
        person.married = False

        return person

    def savePerson(self, person):
        print("Got client params: ")
        print(person.username)
        print(person.age)
        print(person.married)